go/oasis-test-runner/scenario/e2e/txsource.go

```diff
@@ -342,6 +342,17 @@ func (sc *txSourceImpl) startWorkload(childEnv *env.Env, errCh chan error, name
 	return nil
 }
 
+func (sc *txSourceImpl) Clone() scenario.Scenario {
+	return &txSourceImpl{
+		runtimeImpl:           *sc.runtimeImpl.Clone().(*runtimeImpl),
+		workloads:             sc.workloads,
+		timeLimit:             sc.timeLimit,
+		nodeRestartInterval:   sc.nodeRestartInterval,
+		livenessCheckInterval: sc.livenessCheckInterval,
+		rng:                   sc.rng,
```

Comment:
It doesn't seem right to copy the RNG as it can be in an unspecified state and it will be reinitialized anyway. Maybe you would want to copy the seed instead and make sure to reuse the seed if it's already set?

Reply:
I left `rng` uninitialized now and clone `sc.seed` string. In `PreInit()` the seed is now generated only, if it's not set yet.